### ISSUE TITLE
Bad input to predict n.trees causes crash

### ISSUE THREAD

**Issue by ngreifer** [gbm-developers/gbm#45]
When a `numeric` or `integer` of length 0 is given as an argument to `n.trees` in `predict.gbm()`, it causes a crash (not an error), forcing the reset of the R session. This should fail more gracefully so the error can at least be captured rather than causing a full crash. 

Example (will cause a crash, so do it in its own R session):

```
library(gbm)
data("lalonde", package = "cobalt")

fit <- gbm(treat ~ age + educ + married, 
                 data = lalonde, n.trees = 100)
p <- predict(fit, n.trees = numeric()) #crash for non-NULL of length 0

```

**Comment by bgreenwell** [comment]
Thanks @ngreifer I’ll take a look soon and push a fix! You’re welcome to submit a PR as well!

**Comment by bgreenwell** [comment]
Sorry for the massive delay @ngreifer. Fixed in version 2.1.6 (will be on CRAN soon).
